tests/unit_tests/data_steward/retraction/create_deid_map_test.py

```diff
@@ -0,0 +1,125 @@
+import retraction.create_deid_map as create_deid_map
+import utils.bq
+
+
+class CreateDeidMapTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.project_id = 'test_project_id'
+        self.dataset_id = 'test_dataset_id'
+        self.all_datasets = [
+            'R2021Q1R1_combined', 'R2021Q1R1_deid', 'R2021Q1R2_combined',
+            '2021q1r3_combined', 'R2021Q1R3_deid'
+        ]
+        self.deid_datasets = ['R2021Q1R1_deid', 'R2021Q1R3_deid']
+        self.combined_datasets = [
+            'R2021Q1R1_combined', 'R2021Q1R2_combined', '2021q1r3_combined'
+        ]
+
+    @mock.patch('retraction.create_deid_map.utils.bq.list_datasets')
+    def test_get_combined_datasets_for_deid_map(self, mock_list_datasets):
+        result = create_deid_map.get_combined_datasets_for_deid_map(
+            self.project_id)
+
+        expected_all_datasets = mock_list_datasets.return_value = utils.bq.list_datasets(
+            self.project_id)
+        expected_deid_datasets = self.deid_datasets
+        expected_combined_datasets = create_deid_map.get_corresponding_combined_dataset(
+            expected_all_datasets, expected_deid_datasets)
+
+        self.assertListEqual(expected_combined_datasets, result)
+
+    def test_get_corresponding_combined_dataset(self):
+        result = create_deid_map.get_corresponding_combined_dataset(
+            self.all_datasets, self.deid_datasets)
+        expected = list()
+
+        for d in self.deid_datasets:
+            release = d.split('_')[0]
+            older_release = release[1:]
+            combined, older_combined = release + '_combined', older_release + '_combined'
+
+            if combined in self.all_datasets:
+                expected.append(combined)
+            if older_combined in self.all_datasets:
+                expected.append(older_combined)
+            # Remove duplicates
+        expected = list(set(expected))
+
+        self.assertEquals(expected, result)
+
+    @mock.patch('utils.bq.get_table_info_for_dataset')
+    def test_check_if_deid_map_exists(self, mock_get_table_info):
+        result = create_deid_map.check_if_deid_map_exists(
+            self.project_id, self.dataset_id)
+
+        expected_table_info_df = mock_get_table_info.return_value = pd.DataFrame(
+            columns=[
+                'table_catalog', 'table_schema', 'table_name', 'column_name',
+                'ordinal_position', 'is_nullable', 'data_type', 'is_generated',
+                'generation_expression', 'is_stored', 'is_hidden',
+                'is_updatable', 'is_system_defined', 'is_partitioning_column',
+                'clustering_ordinal_position'
+            ])
+        expected_column_list = expected_table_info_df['table_name'].tolist()
```

Comment:
You don't have to define all the columns here. The only column of interest is `table_name`, so the data frame could be instantiated like 
```suggestion
        expected_table_info_df = mock_get_table_info.return_value = pd.DataFrame(
            data={'table_name': ['_deid_map', '_ehr_consent', 'deid_map', common.PERSON, common.OBSERVATION]})
        expected_column_list = expected_table_info_df['table_name'].tolist()
```

Reply:
will update this, thanks!

Reply:
I saw the changes and they make sense. However while you are checking for the assertEquals, you don't need any `if` statements since the conditions are never met for others (no deid_map tables exist in the table name df). It could simply be `self.assertEquals(result, 'CREATE')` and that would be it. 
To test for all the conditions, you'll need to create new dfs with table names having `_deid_map` or `deid_map` or both and use them in side_effects for mock_table info, testing `self.assertEquals(result, action)` separately for each df.